File: backend/app/schemas/coach.py
from pydantic import BaseModel, Field, ConfigDict  
from typing import Optional, Dict, Any, List
from datetime import datetime, date
import uuid

class CoachBase(BaseModel):
    name: str = Field(..., max_length=255)
    full_name: str = Field(..., max_length=255)
    current_club: Optional[str] = None
    current_role: Optional[str] = None
    nationality: Optional[str] = None

class CoachCreate(CoachBase):
    preferred_formation: Optional[str] = None
    date_of_birth: Optional[date] = None
    coaching_level: Optional[str] = None
    years_experience: Optional[int] = None
    estimated_salary_eur: Optional[int] = None
    scouting_notes: Optional[str] = None
    overall_rating: Optional[int] = None
    # potential_rating: Optional[int] = None

class CoachUpdate(CoachCreate):
    # name: Optional[str] = None
    # current_club: Optional[str] = None
    # current_role: Optional[str] = None
    # preferred_formation: Optional[str] = None
    # overall_rating: Optional[int] = Field(None, ge=0, le=100)
    # scouting_notes: Optional[str] = None
    pass

class CoachResponse(CoachBase):
    id: uuid.UUID
    age: Optional[int] = None
    preferred_formation: Optional[str] = None 
    years_experience: Optional[int] = None
    date_of_birth: Optional[date] = None
    estimated_salary_eur: Optional[int] = None
    scouting_notes: Optional[str] = None
    coaching_level: Optional[str] = None
    overall_rating: Optional[int] = None
    # potential_rating: Optional[int] = None
    tactical_style: Optional[Dict[str, Any]] = None
    created_at: datetime
    
    # class Config:
    #     from_attributes = True
    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/types/coach.ts
export interface Coach {
  id: string
  name: string
  full_name?: string
  current_club?: string
  nationality?: string
  date_of_birth?: string
  age?: number
  current_role: string
  preferred_formation: string
  coaching_level: string
  years_experience: number 
  estimated_salary_eur: number
  overall_rating?: number
  // potential_rating?: number
  scouting_notes?: string
  created_at: string
}

export interface CoachFilters {
  current_role?: string
  coaching_level?: string
  nationality?: string
  club?: string
  min_value?: number
  max_value?: number
  search?: string
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}

export interface PaginationMeta {
  page: number
  per_page: number
  total: number
  total_pages: number
}

export interface CoachesResponse {
  success: boolean
  data: Coach[]
  meta: {
    pagination: PaginationMeta
  }
}